src/moments.jl

```diff
@@ -153,66 +153,68 @@ std(v::RealArray, w::AbstractWeights, dim::Int; mean=nothing,
 """
     mean_and_var(x, [w::AbstractWeights], [dim]; corrected=false) -> (mean, var)
 
-Return the mean and variance of a real-valued array `x`, optionally over a dimension
-`dim`, as a tuple. Observations in `x` can be weighted using weight vector `w`.
+Return the mean and standard deviation of collection `x`. If `x` is an `AbstractArray`,
+`dim` can be specified as a tuple to compute statistics over these dimensions.
+A weighting vector `w` can be specified to weight the estimates.
 Finally, bias correction is be applied to the variance calculation if `corrected=true`.
 See [`var`](@ref) documentation for more details.
 """
-function mean_and_var(A::RealArray; corrected::Bool=true)
-    m = mean(A)
-    v = varm(A, m; corrected=corrected)
+function mean_and_var(x; corrected::Bool=true)
+    m = mean(x)
+    v = varm(x, m; corrected=corrected)
     m, v
 end
 
 """
     mean_and_std(x, [w::AbstractWeights], [dim]; corrected=false) -> (mean, std)
 
-Return the mean and standard deviation of a real-valued array `x`, optionally
-over a dimension `dim`, as a tuple. A weighting vector `w` can be specified
-to weight the estimates. Finally, bias correction is applied to the
+Return the mean and standard deviation of collection `x`. If `x` is an `AbstractArray`,
+`dim` can be specified as a tuple to compute statistics over these dimensions.
+A weighting vector `w` can be specified to weight the estimates.
+Finally, bias correction is applied to the
 standard deviation calculation if `corrected=true`.
 See [`std`](@ref) documentation for more details.
 """
-function mean_and_std(A::RealArray; corrected::Bool=true)
-    m = mean(A)
-    s = stdm(A, m; corrected=corrected)
+function mean_and_std(x; corrected::Bool=true)
+    m = mean(x)
+    s = stdm(x, m; corrected=corrected)
     m, s
 end
 
-function mean_and_var(A::RealArray, w::AbstractWeights; corrected::DepBool=nothing)
-    m = mean(A, w)
-    v = varm(A, w, m; corrected=depcheck(:mean_and_var, corrected))
+function mean_and_var(x::RealArray, w::AbstractWeights; corrected::DepBool=nothing)
```

Comment:
Why do we still expect `RealArray` here? I thought this should also work with other arrays. E.g. `Vector{Union{Missing, Float64}}` as long as it does not contain `missing` (which will simply throw an error at run time if it does). Same questions for methods below.

Reply:
Yeah, there are many other things to fix in this file. I'd rather fix them separately in another PR.

Reply:
It is OK to leave it for later. I reacted because you explicitly named this PR "Accept any iterable ..." 😄 so I thought this is exactly the change you want to make in this PR.

Reply:
Yeah the title is a bit too ambitious.